buildSrc/src/main/java/org/elasticsearch/gradle/release/ValidateChangelogsTask.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.release;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.ConfigurableFileCollection;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.util.PatternSet;
+
+import java.io.File;
+
+public class ValidateChangelogsTask extends DefaultTask {
```

Comment:
I'm wondering if it might make sense to use the existing `ValidateJsonAgainstSchemaTask` we use for the REST specs and create a schema for this. I find the error messages produced there pretty good we could move all the logic in `ChangelogEntry.validate()` into something more declarative.